src/YagrCore/plugins/plotLines/plotLines.ts

```diff
@@ -54,12 +49,22 @@ export default function plotLinesPlugin(options: PlotLineOptions): PlotLinesPlug
 
         const hook = HOOKS_MAP[drawIndicies] || 'drawClear';
 
+        function getLineId(line: PlotLineConfig, scale?: string): string {
+            if (line.id) {
+                return line.id;
+            }
+            const lineWithoutId = Array.from(plotLines.entries()).find(
+                ([_, l]) => deepIsEqual(l, line) && (!scale || line.scale === scale),
```

Comment:
why do we need to check ``` !scale || line.scale === scale``` inside the find method? 
If this condition is false we don't even  need to go inside the find method

Reply:
Based on the code, we don't need scale calculations here at all. 
It's only needed in the clear function according to tests.

Reply:
So I removed this logic